From bc705c1e8865f2816bc012f122cb12a5bcf5e754 Mon Sep 17 00:00:00 2001
From: Jan Kiszka <jan.kiszka@siemens.com>
Date: Wed, 27 Jun 2018 12:57:00 +0200
Subject: [PATCH 01/21] can: Fix compiler warning about missing braces

The NULL-test is currently performed also in case we assign a
pre-checked value from the master to base_addr. This is harmless, but
the compiler correctly complains about the misaligned code block - or
the missing braces.

Signed-off-by: Jan Kiszka <jan.kiszka@siemens.com>
---
 kernel/drivers/can/sja1000/rtcan_adv_pci.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/kernel/drivers/can/sja1000/rtcan_adv_pci.c b/kernel/drivers/can/sja1000/rtcan_adv_pci.c
index dc8ea2e..32cf71e 100644
--- a/kernel/drivers/can/sja1000/rtcan_adv_pci.c
+++ b/kernel/drivers/can/sja1000/rtcan_adv_pci.c
@@ -158,14 +158,15 @@ static int rtcan_adv_pci_add_chan(struct pci_dev *pdev,
 			(struct rtcan_adv_pci *)(*master_dev)->board_priv;
 		master_board->slave_dev = dev;
 
-		if (offset)
+		if (offset) {
 			base_addr = master_board->base_addr+offset;
-		else
+		} else {
 			base_addr = pci_iomap(pdev, bar, ADV_PCI_BASE_SIZE);
 			if (!base_addr) {
 				ret = -EIO;
 				goto failure;
 			}
+		}
 	} else {
 		base_addr = pci_iomap(pdev, bar, ADV_PCI_BASE_SIZE) + offset;
 		if (!base_addr) {
-- 
1.9.1

